tests/solvers/average_velocities_01.cc

```diff
@@ -0,0 +1,137 @@
+      simulation_control_parameters.method =
+        Parameters::SimulationControl::TimeSteppingMethod::bdf1;
+      simulation_control_parameters.dt      = 0.1;
+      simulation_control_parameters.timeEnd = 1.0;
+      simulation_control_parameters.adapt   = false;
+
+      // Variables for AverageVelocities
+      AverageVelocities<3, TrilinosWrappers::MPI::Vector, IndexSet> average;
+
+      auto simulation_control = std::make_shared<SimulationControlTransient>(
+        simulation_control_parameters);
+
+      IndexSet locally_owned_dofs;
+      locally_owned_dofs.add_range(0, 3);
+
+      MPI_Comm mpi_communicator = MPI_COMM_WORLD;
+
+      Parameters::PostProcessing postprocessing_parameters;
+      postprocessing_parameters.calculate_average_velocities = true;
+      postprocessing_parameters.initial_time                 = 0.5;
+
+      TrilinosWrappers::MPI::Vector solution(locally_owned_dofs,
+                                             mpi_communicator);
+      solution(0) = 0.0;
+      solution(1) = 2.5;
+      solution(2) = 10;
+
+      TrilinosWrappers::MPI::Vector average_solution(locally_owned_dofs,
+                                                     mpi_communicator);
+
+      // Time info
+      const double time_end     = simulation_control_parameters.timeEnd;
+      const double initial_time = postprocessing_parameters.initial_time;
+      double       time         = simulation_control->get_current_time();
+      double       epsilon      = 1e-6;
+
+      while (time < (time_end + epsilon)) // Until time reached end time
+        {
+          if (time > (initial_time - epsilon)) // Time reached the initial time
+            {
+              average.calculate_average_velocities(solution,
+                                                   simulation_control,
+                                                   postprocessing_parameters,
+                                                   locally_owned_dofs,
+                                                   mpi_communicator);
+
+              average_solution = average.get_average_velocities();
+
+              deallog << " Time :             " << time << std::endl;
+              deallog << " Average solution : " << average_solution[0] << " "
+                      << average_solution[1] << " " << average_solution[2]
+                      << std::endl;
+              deallog << "" << std::endl;
+
+              solution *= 0.9; // new solution values for next step
+            }
+
+          // Integrate to get the next time
+          simulation_control->integrate();
+
+          // Break if the next time from integrate() is the same because
+          // time will never get over the time end, but the average velocities
+          // at this time is wanted.
+          if (abs(time - simulation_control->get_current_time()) < epsilon)
+            break;
+
+          time = simulation_control->get_current_time();
+        }
```

Comment:
It is better to put the test itself in a test function instead of directly in the main file.

Reply:
It means line 71 to 107 should be in a function?